frontend/components/inputs/UploadTimeSeriesFileInput.js

```diff
@@ -0,0 +1,181 @@
+import {getCookie} from "../../common";
+import PropTypes from "prop-types";
+
+
+const UploadTimeSeriesFileInput = ({id, uploadSuccessfulCallback, apiEndpoint}) => {
+    const [submitted, setSubmitted] = useState({"file": false});
+    const [tempFile, setTempFile] = useState(null);
+    const [constants, setConstants] = useState(
+        [
+            {
+                "base_frequency": 100,
+                "multiplier": 20,
+                "offset": 0,
+                "a_percentage": .25,
+                "d_percentage": 0,
+                "s_percentage": .5,
+                "r_percentage": .25,
+            },
+            {
+                "base_frequency": 150,
+                "multiplier": 20,
+                "offset": 0,
+                "a_percentage": .2,
+                "d_percentage": .4,
+                "s_percentage": .3,
+                "r_percentage": .1,
+            },
+            {
+                "base_frequency": 225,
+                "multiplier": 20,
+                "offset": 0,
+                "a_percentage": .2,
+                "d_percentage": .4,
+                "s_percentage": .3,
+                "r_percentage": .1,
+            },
+        ]
+    );
+
+    const fileRef = useRef(null);
+    const handleSubmitFile = (event) => {
+        event.preventDefault();
+        setSubmitted(prevSubmitted => ({...prevSubmitted, "file": true}));
+        submitFileToAPI(tempFile, "file");
+    };
+
+    const handleFileInput = (event) => {
+        setTempFile(event.target.files[0]);
+    };
+
+    const updateConstant = (col, name, val) => {
+        setConstants(prevState => {
+            let temp = Object.assign([], prevState);
+            temp[col][name] = parseFloat(val);
+            return temp;
+        });
+    };
+
+    const clearFile = () => {
+        const file = fileRef.current;
+        file.value = null;
+        setTempFile(null);
+        setSubmitted(prevSubmitted => ({...prevSubmitted, "file": false}));
+    };
+
+    const submitFileToAPI = (file) => {
+        const formData = new FormData();
+        formData.append("type", "file");
+        formData.append("id", id);
+        formData.append("value", file, "tempfile.csv");
+        formData.append("constants", JSON.stringify(constants));
+        const csrftoken = getCookie("csrftoken");
+        const requestOptions = {
+            method: "POST",
+            headers: {
+                "X-CSRFToken": csrftoken,
+            },
+            body: formData,
+        };
+        fetch(apiEndpoint, requestOptions)
+            .then(response => response.json())
+            .then(data => uploadSuccessfulCallback(data));
+    };
+
+    return (
+        <>
+            <div>
+                {constants.map((each, i) => {
+                    return <div className="form-inline" key={i}>
+                        <p>Column {i}: </p>
+                        <div>
+                            <label>Offset</label>
```

Comment:
There is a lot of repetitive code here! I am fine merging it in this state, but with the `todo` soon that this will turn into inputs generated on the fly in response to the upload of the csv, if that is the intention.